charts/primary-site/templates/deployments/inbox-listener.yaml

```diff
@@ -1,3 +1,38 @@
+{{- if .Values.inboxListener.autoscaling.enabled }}
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: keda-inbox-listener-auth
+spec:
+  secretTargetRef:
+    - parameter: token
+      name: foxglove-site-token
+      key: FOXGLOVE_SITE_TOKEN
+---
+apiVersion: keda.sh/v1alpha1
+kind: ScaledJob
+metadata:
+  name: inbox-listener-scaledjob
+spec:
+  minReplicaCount: {{ .Values.inboxListener.autoscaling.minReplicas }}
+  maxReplicaCount: {{ .Values.inboxListener.autoscaling.maxReplicas }}
+  successfulJobsHistoryLimit: 50
+  failedJobsHistoryLimit: 100
+  pollingInterval: 30
+  jobTargetRef:
+    parallelism: 1
+    activeDeadlineSeconds: 86400
+    {{ include "primary-site.inbox-container" . | nindent 4 }}
+  triggers:
+    - type: metrics-api
+      metadata:
+        targetValue: "2"
+        url: "{{ .Values.globals.foxgloveApiUrl }}/internal/platform/v1/pending-imports-stats"
```

Comment:
Suddenly this internal endpoint doesn't feel as internal. Could this feasibly be read from the site controller at startup? Could the site controller itself proxy these requests to the API?

Reply:
Maybe another option is to have the site-controller serve its cached metric directly; it's already polling this endpoint in the background and has a server running for liveness checks.